### PR TITLE
feat(ST-60): implement teacher profile interface

### PR DESCRIPTION
**Pull request by akib-raihan-sazim**
## Description of Change
- Updated `user.api.ts` file for fetching user's details
- Updated `Navbar.tsx` to navigate to profile page
- Created Page for profile
- Created `ProfileContainer` 
- Created `TeacherProfile` component
- Update `user.types.ts` to include user, student and teacher types.

## Associated Ticket Link(s)
- https://trello.com/c/gCEsdU3F

## Screenshots / Screen Recordings

https://github.com/user-attachments/assets/758b2e86-93fd-49f5-9031-b25ef2d5a50a

### CODE REVIEW

File: modules/Profile/components/TeacherProfile/TeacherProfile.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import { ITeacherProfileProps } from "./TeacherProfile.interface";
+
+const TeacherProfile: React.FC<ITeacherProfileProps> = ({ user }) => {
+  if (!user || !user.teacher) {
+    notifications.show({
+      title: "Profile Data Unavailable",
+      message: "No teacher profile data is available at the moment.",
+      color: "red",
+    });
+  }
```
Comment (ishmam-iqbal-sazim) [inline modules/Profile/components/TeacherProfile/TeacherProfile.tsx:13]: Do you need this? Isn't this already being handled in the ProfileContainer

File: modules/Profile/components/TeacherProfile/TeacherProfile.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+  if (!user || !user.teacher) return null;
```
Comment (ishmam-iqbal-sazim) [inline modules/Profile/components/TeacherProfile/TeacherProfile.tsx:15]: Same comment as above

File: modules/Profile/components/TeacherProfile/TeacherProfile.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  ));
+
+  const profileItems = [
+    { label: "Email", value: user.email },
+    { label: "Gender", value: user.gender || "Not specified" },
+    { label: "First name", value: user.firstName },
+    { label: "Last name", value: user.lastName },
+    { label: "Major Subject", value: user.teacher.majorSubject },
+    { label: "Highest education level", value: user.teacher.highestEducationLevel },
+  ];
```
Comment (ishmam-iqbal-sazim) [inline modules/Profile/components/TeacherProfile/TeacherProfile.tsx:39]: This looks like a helper that takes in user as param. Should be inside the .helpers.ts file

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments

**Review by ragib-hossain-sazim — APPROVED** [review]

File: modules/Profile/components/TeacherProfile/TeacherProfile.tsx (hunk 2 above)
Comment (ishmam-iqbal-sazim) [reply modules/Profile/components/TeacherProfile/TeacherProfile.tsx]: 1. Should the user be optional if you are checking for the user in ProfileContainer anyway?
2. Aren't we already checking if user is a teacher in ProfileContainer before rendering this component?

**Pull request merged**